vllm/entrypoints/openai/protocol.py

```diff
@@ -59,10 +62,34 @@ class ResponseFormat(BaseModel):
     type: str = Literal["text", "json_object"]
 
 
+class TextContent(BaseModel):
+    type: str = Literal["text"]
+    text: str
+
+
+SingleTextContentList = Annotated[List[TextContent], Len(1, 1)]
+
+
+def try_flatten_text_content(value: Any) -> str:
+    if not isinstance(value, str):
+        # The OpenAI API allows complex contents in chat messages. While
+        # we do not fully support them at the moment, we can allow simple
+        # cases where just a single text content is passed in a message.
+        with contextlib.suppress(ValidationError):
+            single_text_content_list = TypeAdapter(
+                SingleTextContentList).validate_python(value)
+            return single_text_content_list[0].text
+    return value
+
+
+ChatCompletionMessage = Dict[str, Annotated[
+    str, BeforeValidator(try_flatten_text_content)]]
```

Comment:
still a bit unsatisfied that we cannot just do Union[str, List[TextContent]] and then turn the `TextContent` into `str` in the post processing in `ChatCompletionRequest`.

Reply:
OK, I have moved the flattening to a dedicated method. Let me know what you think @simon-mo 🙇